tgui/packages/tgui/interfaces/Mecha/ModulesPane.tsx

```diff
@@ -477,16 +490,46 @@ const SnowflakeSyringe = (props) => {
         />
       </LabeledList.Item>
       <LabeledList.Item label={'Reagent control'}>
+        {analyzed_reagents.map((reagent) => (
+          <LabeledList.Item key={`${reagent}`} label={`${reagent}`}>
+            <Button
+              content={'Toggle Synthesizing'}
+              onClick={() =>
+                act('equip_act', {
+                  ref: ref,
+                  gear_action: `toggle_reagent_${reagent}`,
+                })
+              }
+            />
+          </LabeledList.Item>
+        ))}
+      </LabeledList.Item>
+      <LabeledList.Item>
         <Button
-          content={'View'}
+          content={'Purge All'}
           onClick={() =>
             act('equip_act', {
               ref: ref,
-              gear_action: 'show_reagents',
+              gear_action: `purge_all`,
             })
           }
         />
       </LabeledList.Item>
+      {contained_reagents.map((reagent: Reagent) => (
```

Comment:
props should be typed so you wouldn't need to type reagent here. im assuming there are other any types here